srcDLL/DllMain.cpp

```diff
@@ -143,12 +141,13 @@ void LocateVolFiles(const std::string& relativeDirectory)
 			const auto extension = ToLower(filePath.extension().string());
 
 			if (extension == ".vol") {
-				volList->AddVolFile((fs::path(relativeDirectory) / filePath.filename()).string());
+				auto relativePath = fs::relative(absoluteDirectory, GetExeDirectory()) / filePath.filename();
+				volList->AddVolFile(relativePath.string());
 			}
 		}
 	}
 	catch (const fs::filesystem_error& e) {
-		Log(e.what());
+		Log("Unable to locate vol files in provided directory " + absoluteDirectory + ". " + std::string(e.what()));
```

Comment:
This message only contains information available to the caller. Perhaps the caller should catch and log exceptions.

Reply:
I moved this function to within VolList. I think catching the exception there makes sense. Let me know if you still want it more generalized.